### PR TITLE
Provide all (relevant runs of all) jobs in deployment API

### PR DESCRIPTION
**Pull request by jonmv**
@freva please review.

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/DeploymentApiHandler.java [hunk 1]
```diff
@@ -129,25 +124,73 @@ private HttpResponse root(HttpRequest request) {
...
+                      instanceObject.setString("instance", instance.instance().value());
+                      instanceObject.setLong("productionJobCount", jobsByInstance.get(instance).production().size());
+                      instanceObject.setString("upgradePolicy", toString(deploymentStatuses.matching(status -> status.application().id().equals(TenantAndApplicationId.from(instance)))
+                                                                                           .first().map(status -> status.application().deploymentSpec())
+                                                                                           .flatMap(spec -> spec.instance(instance.instance()).map(DeploymentInstanceSpec::upgradePolicy))
+                                                                                           .orElse(DeploymentSpec.UpgradePolicy.defaultPolicy)));
+                      Cursor allJobsObject = instanceObject.setObject("allJobs");
+                      Cursor upgradeJobsObject = instanceObject.setObject("upgradeJobs");
+                      for (JobType type : JobType.allIn(controller.system())) {
```
Comment (freva) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/DeploymentApiHandler.java:171]: Would be nice if we could just list relevant jobs, and maybe list all the jobs once in a list for order, if we need that.

**Review by freva — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/DeploymentApiHandler.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/deployment/DeploymentApiHandler.java:171]: I guess that's an option that saves some JSON over the wire :)

**Comment by jonmv** [comment]
PTAL.

**Review by freva — APPROVED** [review]
:100:

**Pull request merged**